tests/test_curobo.py

```diff
@@ -26,93 +37,427 @@ def test_curobo():
+            target_quat[robot.eef_link_names[arm_name]] = th.stack(target_quat[robot.eef_link_names[arm_name]], dim=0)
+            target_pos_in_world_frame[robot.eef_link_names[arm_name]] = th.stack(
+                target_pos_in_world_frame[robot.eef_link_names[arm_name]], dim=0
+            )
+
+        # Cast defaultdict to dict
+        target_pos = dict(target_pos)
+        target_quat = dict(target_quat)
+        target_pos_in_world_frame = dict(target_pos_in_world_frame)
+
+        print(f"Planning for {len(target_pos[robot.eef_link_names[robot.default_arm]])} eef targets...")
+
+        # Generate collision-free trajectories to the sampled eef poses (including self-collisions)
+        successes, traj_paths = cmg.compute_trajectories(
+            target_pos=target_pos,
+            target_quat=target_quat,
+            is_local=True,
+            max_attempts=1,
+            timeout=60.0,
+            ik_fail_return=5,
+            enable_finetune_trajopt=True,
+            finetune_attempts=1,
+            return_full_result=False,
+            success_ratio=1.0,
+            attached_obj=None,
+        )
+
+        # Make sure collision-free trajectories are generated
+        success_rate = successes.double().mean().item()
+        print(f"Collision-free trajectory generation success rate: {success_rate}")
+        assert success_rate == 1.0, f"Collision-free trajectory generation success rate: {success_rate}"
+
+        # 1cm and 3 degrees error tolerance for prismatic and revolute joints, respectively
+        error_tol = th.tensor(
+            [0.01 if joint.joint_type == "PrismaticJoint" else 3.0 / 180.0 * math.pi for joint in robot.joints.values()]
+        )
+
+        for bypass_physics in [True, False]:
+            for traj_idx, (success, traj_path) in enumerate(zip(successes, traj_paths)):
+                if not success:
+                    continue
+
+                # Reset the environment
+                env.scene.reset()
+
+                # Move the markers to the desired eef positions
+                for marker, arm_name in zip(eef_markers, robot.arm_names):
+                    eef_link_name = robot.eef_link_names[arm_name]
+                    marker.set_position_orientation(position=target_pos_in_world_frame[eef_link_name][traj_idx])
+
+                q_traj = cmg.path_to_joint_trajectory(traj_path)
+                # joint_positions_set_point = []
+                # joint_positions_response = []
+                for i, q in enumerate(q_traj):
+                    if bypass_physics:
+                        print(f"Teleporting waypoint {i}/{len(q_traj)}")
+                        robot.set_joint_positions(q)
+                        robot.keep_still()
+                        og.sim.step()
+                        for contact in robot.contact_list():
+                            assert contact.body0 in robot.link_prim_paths
```

Comment:
is this always the order (body0 in robot link prim paths, instead of body1)?

Reply:
Yes.